### DISCUSSION TITLE
Extract both texts and tables and images on the same page

### DISCUSSION THREAD

**Discussion by washyouself** [jsvine/pdfplumber#1105] [Q&A]
I am a beginner who uses pdfplumber, I hope to extract text and tables and images on the same page in natural order, what should I do? Is there any code example for reference? All the APIS I've seen are extracting some sort of element, all the text, all the forms, all the images. 

I want to be able to get a page object and parse the pdf in a natural order.

**Comment by jsvine** [comment]
The tricky bit here is "natural order" — an order may seem natural to a human, but isn't necessarily encoded in the PDF. "Natural order" logic can probably be custom coded for specific types of PDFs, but that particular logic will depend on how those PDFs are arranged. 

There's also a question of "extract _to what_?", since tables have a different sort of representation than plain text and than images.
